server/protocols/protocols-imap4/src/main/java/org/apache/james/imapserver/netty/IMAPServerFactory.java

```diff
@@ -43,28 +45,32 @@ public class IMAPServerFactory extends AbstractServerFactory {
     protected final ThrowingFunction<HierarchicalConfiguration<ImmutableNode>, ImapSuite> imapSuiteProvider;
     protected final ImapMetrics imapMetrics;
     protected final GaugeRegistry gaugeRegistry;
+    protected final ThrowingFunction<HierarchicalConfiguration<ImmutableNode>, Set<ConnectionCheck>> connectionCheckProvider;
 
     @Inject
     @Deprecated
     public IMAPServerFactory(FileSystem fileSystem, ImapDecoder decoder, ImapEncoder encoder, ImapProcessor processor,
-                             MetricFactory metricFactory, GaugeRegistry gaugeRegistry) {
+                             MetricFactory metricFactory, GaugeRegistry gaugeRegistry, ThrowingFunction<HierarchicalConfiguration<ImmutableNode>, Set<ConnectionCheck>> connectionCheckProvider) {

```

Comment:
```suggestion
                             MetricFactory metricFactory, GaugeRegistry gaugeRegistry, Function<HierarchicalConfiguration<ImmutableNode>, Set<ConnectionCheck>> connectionCheckProvider
```

That's lighter. Tips: Sneaky throw.

Or even better: name this explicitly with an interface:

```
@FunctionalInterface
interface ConnectionCheckFactory {
    Set<ConnectionCheck> create(ImapConfiguration imapConfiguration)
}
```

BTW that's mabe a refactoring but we would need to convert the XML into an ImapConfiguration in this class as soon as we can.

Reply:
You mean that parse imapserver.xml to get set of ConnectionCheck, then add them into  ImapConfig?

Reply:
Let's put `additionalConnectionChecks` into `ImapConfiguration`. And init `Set<ConnectionCheck>` from `ImapConfiguration`, as Benoit suggestion.